presto-main/src/main/java/io/prestosql/sql/planner/PlanOptimizers.java

```diff
@@ -623,7 +624,8 @@ public PlanOptimizers(
                         new CreatePartialTopN(),
                         new PushTopNThroughProject(),
                         new PushTopNThroughOuterJoin(),
-                        new PushTopNThroughUnion())));
+                        new PushTopNThroughUnion(),
+                        new PushTopNIntoTableScan(metadata))));
```

Comment:
This should be added to `pushIntoTableScanOptimizer`, when other `PushXxxIntoTableScan` rules are registered.

This is important for two reasons:
- this will unlock other pushdowns once TopN is pushed into TableScan fully
- this will let the rule operate on TopNNode while it is SINGLE step (and consume it fully), making reasoning about engine-connector interactions simpler

@wendigo is going to address this in https://github.com/trinodb/trino/pull/6847